test/core/ExamplesUnitTests/TribitsExampleProject2_Tests.cmake

```diff
@@ -1068,3 +1068,297 @@ TribitsExampleProject2_External_Package_by_Package(SHARED  CMAKE_PREFIX_PATH_CAC
 
 # NOTE: The above tests check a few different use cases for building and
 # installing TriBITS packages from a single TriBITS project incrementally.
+
+
+################################################################################
+
+
+function(TribitsExampleProject2_External_RawPackage1_then_Package_by_Package
+    sharedOrStatic  findingTplsMethod
+  )
+
+  TribitsExampleProject2_test_setup_header()
+
+  set(tplInstallBaseDir
+    "${TribitsExampleProject2_Tpls_install_${sharedOrStatic}_DIR}")
+
+  set(allTplsNoPrefindArgs
+    "-DTpl1_ALLOW_PACKAGE_PREFIND=OFF"
+    "-DTpl2_ALLOW_PACKAGE_PREFIND=OFF"
+    "-DTpl3_ALLOW_PACKAGE_PREFIND=OFF"
+    "-DTpl4_ALLOW_PACKAGE_PREFIND=OFF"
+    )
+
+  if (sharedOrStatic STREQUAL "STATIC")
+    set(libExt "a")
+  elseif (sharedOrStatic STREQUAL "SHARED")
+    set(libExt "so")
+  else()
+    message(FATAL_ERROR "Error: Invalid value of"
+      " sharedOrStatic='${sharedOrStatic}'!")
+  endif()
+
+  if (findingTplsMethod STREQUAL "TPL_LIBRARY_AND_INCLUDE_DIRS")
+    set(tpl1LibAndIncDirsArgs
+      "-DTpl1_INCLUDE_DIRS=${tplInstallBaseDir}/install_tpl1/include"
+      "-DTpl1_LIBRARY_DIRS=${tplInstallBaseDir}/install_tpl1/lib"
+      "-DTpl1_ALLOW_PACKAGE_PREFIND=OFF")
+    set(tpl2LibAndIncDirsArgs
+      "-DTpl2_INCLUDE_DIRS=${tplInstallBaseDir}/install_tpl2/include"
+      "-DTpl2_LIBRARY_DIRS=${tplInstallBaseDir}/install_tpl2/lib"
+      "-DTpl2_ALLOW_PACKAGE_PREFIND=OFF")
+    set(tpl3LibAndIncDirsArgs
+      "-DTpl3_INCLUDE_DIRS=${tplInstallBaseDir}/install_tpl3/include"
+      "-DTpl3_LIBRARY_DIRS=${tplInstallBaseDir}/install_tpl3/lib"
+      "-DTpl3_ALLOW_PACKAGE_PREFIND=OFF")
+    set(tpl4LibAndIncDirsArgs
+      "-DTpl4_INCLUDE_DIRS=${tplInstallBaseDir}/install_tpl4/include"
+      "-DTpl4_ALLOW_PACKAGE_PREFIND=OFF")
+    set(tpl1CMakePrefixPath "")
+    set(tpl2CMakePrefixPath "")
+    set(tpl3CMakePrefixPath "")
+    set(tpl4CMakePrefixPath "")
+    set(tpl1FoundRegexes
+      "TPL_Tpl1_LIBRARIES='.*/install_tpl1/lib/libtpl1[.]${libExt}'"
+      "TPL_Tpl1_INCLUDE_DIRS='.*/install_tpl1/include'")
+    set(tpl2FoundRegexes
+      "TPL_Tpl2_LIBRARIES='.*/install_tpl2/lib/libtpl2b[.]${libExt}[;].*/install_tpl2/lib/libtpl2a[.]${libExt}'"
```

Comment:
Interesting to wrap the `;` in brackets. Does `tribits_add_advanced_test()` know to handle this as part of the regex, or does it naively split on semicolons, with the first one ending with `[` and the second starting with `]`?

Reply:
`tribits_add_advanced_test()` splits the list of regexes as usual.  It is just that CMake does not split a string on `;` if you enclose it in `[;]`.